src/main/java/io/confluent/connect/jdbc/source/JdbcSourceConnectorConfig.java

```diff
@@ -587,6 +615,77 @@ public boolean visible(String name, Map<String, Object> config) {
     }
   }
 
+  public enum NumericMapping {
+    NONE,
+    ONLY_PRECISION,
+    COMPLETE;
+
+    private static final Map<String, NumericMapping> reverse = new HashMap<>(values().length);
+    static {
+      for (NumericMapping val : values()) {
+        reverse.put(val.name().toLowerCase(Locale.ROOT), val);
+      }
+    }
+
+    public static NumericMapping get(String prop) {
+      // not adding a check for null value because the recommender/validator should catch those.
+      return reverse.get(prop);
+    }
+
+    public static NumericMapping get(JdbcSourceConnectorConfig config) {
+      if (config.getBoolean(JdbcSourceTaskConfig.NUMERIC_PRECISION_MAPPING_CONFIG)) {
+        return NumericMapping.ONLY_PRECISION;
+      }
+      return get(config.getString(JdbcSourceConnectorConfig.NUMERIC_MAPPING_CONFIG));
+    }
+  }
+
+  //Porting from JdbcSinkConfig and extending to implement Recommender interface too.
+  //TODO: Should factor out to common class.
+  private static class EnumRecommender implements ConfigDef.Validator, ConfigDef.Recommender {
+    private final List<String> canonicalValues;
+    private final Set<String> validValues;
+
+    private EnumRecommender(List<String> canonicalValues, Set<String> validValues) {
+      this.canonicalValues = canonicalValues;
+      this.validValues = validValues;
+    }
+
+    public static <E> EnumRecommender in(E[] enumerators) {
+      final List<String> canonicalValues = new ArrayList<>(enumerators.length);
+      final Set<String> validValues = new HashSet<>(enumerators.length * 2);
+      for (E e : enumerators) {
+        canonicalValues.add(e.toString().toLowerCase());
+        validValues.add(e.toString().toUpperCase(Locale.ROOT));
+        validValues.add(e.toString().toLowerCase(Locale.ROOT));
+      }
+      return new EnumRecommender(canonicalValues, validValues);
+    }
+
+    @Override
+    public void ensureValid(String key, Object value) {
+      // calling toString on itself because IDE complains if the Object is passed.
+      if (!validValues.contains(value.toString())) {
+        throw new ConfigException(key, value, "Invalid enumerator");
```

Comment:
Maybe reframe a message to something like `Invalid enumeration value`

Reply:
Keeping the old one to match what we have in `JdbcSinkConfig` (I added a comment to mention this comes from there).